### ISSUE TITLE
Mismatch between Route Coordination and Forward RFC

### ISSUE THREAD

**Issue by rolsonquadras** [hyperledger/aries-framework-go#965]
The recKeys are registered on the router as per the [Route Coordination RFC](https://github.com/hyperledger/aries-rfcs/tree/master/features/0211-route-coordination#keylist-update), for which the edge agents expect redirection/forward from the router. But the `to` field in the [Forward message](https://github.com/hyperledger/aries-rfcs/blob/master/concepts/0094-cross-domain-messaging/README.md#corerouting10forward) takes a destination DID with/without key references. The Router won't be able to figure out the recKey from DID.

**Comment by rolsonquadras** [comment]
@llorllale @troyronda created this to track our earlier discussion.

**Comment by troyronda** [comment]
This issue seems like it could be an aries-rfcs discussion (issue)?

**Comment by llorllale** [comment]
hyperledger/aries-rfcs#274

**Comment by rolsonquadras** [comment]
To bridge the gap between the [Route Coordination RFC](https://github.com/hyperledger/aries-rfcs/tree/master/features/0211-route-coordination#keylist-update) and  [Forward message](https://github.com/hyperledger/aries-rfcs/blob/master/concepts/0094-cross-domain-messaging/README.md#corerouting10forward), the Aries Framework Go uses Recipient key in the `to` field of forward message instead of DID.

Forward RFC:
[`"to"   : "did:sov:1234abcd#4",`](https://github.com/hyperledger/aries-rfcs/blob/master/concepts/0094-cross-domain-messaging/README.md#corerouting10forward)

Aries Framework Go Implementation:
[`"to"   : "<recKey of the destination agent>",`](https://github.com/hyperledger/aries-framework-go/blob/ee769a304613c3dc3fb44edad780d674921d567a/pkg/didcomm/dispatcher/outbound.go#L154)

**Comment by troyronda** [comment]
> To bridge the gap

(as a temporary workaround pending further discussions).

**Comment by rolsonquadras** [comment]
> > To bridge the gap
> 
> (as a temporary workaround pending further discussions).

Yes, @llorllale had raised this question in Aries RFC issue https://github.com/hyperledger/aries-rfcs/issues/274#issuecomment-565172057.

**Comment by TelegramSam** [comment]
I'm realizing that this RFC missed an update to reflect an early decision. We ended up only using keys for routing because it was the simplest thing that would work. On that note, the decision in Aries Framework Go follows that method.

On the plus side, now is a really good time to revisit this issue.
